ta/pkcs11/src/attributes.c

```diff
@@ -207,6 +207,50 @@ bool get_bool(struct obj_attrs *head, uint32_t attribute)
 	return bbool;
 }
 
+bool attributes_match_reference(struct obj_attrs *candidate,
+				struct obj_attrs *ref)
+{
+	size_t count = ref->attrs_count;
+	unsigned char *ref_attr = ref->attrs;
+	uint32_t rc = PKCS11_CKR_GENERAL_ERROR;
+
+	if (!ref->attrs_count) {
+		DMSG("Empty reference: no match");
+		return false;
+	}
+
+	for (count = 0; count < ref->attrs_count; count++) {
+		struct pkcs11_attribute_head pkcs11_ref;
+		void *found = NULL;
+		uint32_t size = 0;
+		int shift = 0;
+
+		TEE_MemMove(&pkcs11_ref, ref_attr, sizeof(pkcs11_ref));
+
+		shift = pkcs11_attr2boolprop_shift(pkcs11_ref.id);
```

Comment:
This boolprop stuff is in @etienne-lms's branch as conditional code but I am not yet convinced that it is actually a good thing to have. It causes problems for serializers. I understand that it makes things slightly more compact but also changes storage and REE interfaces where as storage is more challenging to change.

But the code what you want to do here is actually _is_attribute_bool()_ which could internally use whatever method is in use so shall we have helper for that?

Reply:
Makes sense. I will introduce a helper function

Reply:
Agree. Thank @vesajaaskelainen.